addons/mass_mailing/models/mail_mail.py

```diff
@@ -26,8 +26,8 @@ def create(self, values_list):
 
     def _get_tracking_url(self):
         base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
-        track_url = werkzeug.urls.url_join(base_url, 'mail/track/%s/blank.gif' % self.id)
-        return '<img src="%s" alt=""/>' % track_url
+        token = tools.hmac(self.env(su=True), 'mass_mailing-mail_mail-open', self.id)
```

Comment:
@odony As the URL is not visible, there's no reason to truncate it IMO
But good idea for the route :)

Reply:
My point is more about the unnecessary wasted bytes and ugliness of the URL, not because it is visible to users. But it may become a broader concern if we start using sha-256 hmacs for other things that will be visible in URLs.
Anyways, if you prefer the long version, fine ;-)